tests/conftest.py

```diff
@@ -1575,6 +1575,23 @@ def patch(compile: data.Compile, project_dir: str) -> CompileRun:
         return CompileRunnerMock(compile, make_compile_fail, runner_queue)
 
     monkeypatch.setattr(compilerslice, "_get_compile_runner", patch, raising=True)
+    monkeypatch.setattr(compilerslice, "_running_compiles", {}, raising=False)
```

Comment:
This could do with a comment.

Reply:
I had to add this because I made the `_recompiles` dict into the `_compiling_envs` set. Since @arnaudsjs just had to change it back, I will revert this.